server/services/ast/inspector.ts

```diff
@@ -46,14 +47,19 @@ function getTemplateLiteralName(path: any) {
         `\`\$\{${
           path.node.arguments[0].expressions[currentExpressionIndex++].name
         }\}\``
-      );
+        );
     } else {
       return finalText.concat(q.value.raw);
     }
   }, "");
 }
+function getNodeName(path: any): string {
```

Comment:
This is a neat refactor 👍

Reply:
Yea, I don't like repeating code - especially in JS/TS where it doesn't get compiled away. And once I realized that it could be collapsed like that, I felt it was easier to understand what is happening. 

[Warning: off topic :) ]
One thing I have learned over the years is that it is much easier to understand code when it can fit on one screen. So I break things into smaller functions with describtive names. So instead of having a huge function, I will break things up like this: 

```JS
function () {
  for(var element of array() ) {
    doStep1OfAlgorthm(...); // maybe this a 100 line function
    doStep2OfAlgorithm(...); // maybe another 100 line function
   if(success) {
      doCommitOfAlgorithm(...);
    }
  }
}
```

Since I can see the entire loop on the screen at once, I can figure out what the function is doing, once it gets too big to fit on the screen, it becomes much harder to work with. 
Sorry for the divergence, I just wanted to share a bit of what I learned the hard way over the years. It has made my code much cleaner and easier to understand. And it was partially what drove this refactor. I can just get findItems on one screen now. I might take the refactoring further and collapse the setup portion into a function called:

parseTestItemAttributes(...)
that returns an object of type:string and only: boolean.

Then I might collapse the action portion into a function called:
addTestItemToResultList(...)

that way the findItems function becomes: 
```JS
function findItems(path: any, result: TestItem[], idManager: IdManager, parentId?: any) {
  var pathAttributes = parseTestItemAttributes(path);
  addTestItemToResultList(path, pathAttributes);
}
```
but I didn't want to introduce too much change into a code base that isn't mine.